packages/core/src/oidc/extra-token-claims.ts

```diff
@@ -125,22 +119,28 @@ export const getExtraTokenClaimsForJwtCustomization = async (
         (await libraries.jwtCustomizers.getUserContext(token.accountId))
     );
 
-    // `context` parameter is only eligible for user's access token for now.
-    return await client.post(`/api/services/custom-jwt`, {
-      body: isTokenClientCredentials
-        ? {
-            ...commonPayload,
-            tokenType: LogtoJwtTokenKeyType.ClientCredentials,
-          }
-        : {
-            ...commonPayload,
-            tokenType: LogtoJwtTokenKeyType.AccessToken,
-            // TODO (LOG-8555): the newly added `UserProfile` type includes undefined fields and can not be directly assigned to `Json` type. And the `undefined` fields should be removed by zod guard.
-            // eslint-disable-next-line no-restricted-syntax
-            context: { user: logtoUserInfo as Record<string, Json> },
-          },
-      search: {},
-    });
+    const payload: CustomJwtFetcher = isTokenClientCredentials
+      ? {
+          ...commonPayload,
+          tokenType: LogtoJwtTokenKeyType.ClientCredentials,
+        }
+      : {
+          ...commonPayload,
+          tokenType: LogtoJwtTokenKeyType.AccessToken,
+          // TODO (LOG-8555): the newly added `UserProfile` type includes undefined fields and can not be directly assigned to `Json` type. And the `undefined` fields should be removed by zod guard.
+          // eslint-disable-next-line no-restricted-syntax
+          context: { user: logtoUserInfo as Record<string, Json> },
+        };
+
+    if (EnvSet.values.isCloud) {
+      const client = await cloudConnection.getClient();
+      // `context` parameter is only eligible for user's access token for now.
+      return await client.post(`/api/services/custom-jwt`, {
+        body: payload,
+        search: {},
```

Comment:
no need for this?

Reply:
Should keep this for now, if we mark `serach` field to be optional in Cloud, the field can not even be available for the withtyped client.